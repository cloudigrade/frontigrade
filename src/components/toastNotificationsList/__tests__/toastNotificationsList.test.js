import React from 'react';
import configureMockStore from 'redux-mock-store';
import { shallow, mount } from 'enzyme';
import { ConnectedToastNotificationsList, ToastNotificationsList } from '../toastNotificationsList';

describe('ToastNotificationsList Component', () => {
  const generateEmptyStore = (obj = {}) => configureMockStore()(obj);

  it('should render a basic component', () => {
    const store = generateEmptyStore({
      toastNotifications: {
        toasts: [
          { removed: true, alertType: 'success', header: 'lorem', message: 'lorem' },
          { removed: false, alertType: 'success', header: 'ipsum', message: 'ipsum' }
        ],
        displayedToasts: 1
      }
    });
    const component = shallow(<ConnectedToastNotificationsList />, { context: { store } });

    expect(component.render()).toMatchSnapshot();
    expect(component.render().find('.toast-pf').length).toEqual(1);
  });

  it('should have specific events defined', () => {
    const component = mount(<ToastNotificationsList />);
    const componentInstance = component.instance();

    expect(componentInstance.onHover).toBeDefined();
    expect(componentInstance.onLeave).toBeDefined();
    expect(componentInstance.onDismiss).toBeDefined();
  });
});
