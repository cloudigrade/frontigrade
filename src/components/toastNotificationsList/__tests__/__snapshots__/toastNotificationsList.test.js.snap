// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToastNotificationsList Component should render a basic component 1`] = `
<div
  class="toast-notifications-list-pf"
>
  <div
    class="alert toast-pf alert-success alert-dismissable"
  >
    <button
      aria-hidden="true"
      class="close close-default"
      type="button"
    >
      <span
        aria-hidden="true"
        class="pficon pficon-close"
      />
    </button>
    <span
      aria-hidden="true"
      class="pficon pficon-ok"
    />
    <span>
      <strong>
        ipsum
      </strong>
       Â ipsum
    </span>
  </div>
</div>
`;
